pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -247,6 +259,43 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	return nil
 }
 
+func finalizeIfNeeded(ctx context.Context, stream *imagev1.ImageStream, srcClusterName string, clients map[string]ctrlruntimeclient.Client) error {
+	finalizerSet := sets.NewString(stream.ObjectMeta.Finalizers...)
+	if finalizerSet.Has(ControllerName) {
+		for clusterName, client := range clients {
+			if clusterName == srcClusterName {
+				continue
+			}
+			// populate deleting to all clusters except srcClusterName
+			if err := client.Delete(ctx, &imagev1.ImageStream{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      stream.Name,
+					Namespace: stream.Namespace,
+				},
+			}); !apierrors.IsNotFound(err) {
```

Comment:
This is wrong, because if the error is nil we will return here without removing the finalizer, needs to be `err != nil && !apierrors.IsNotFound(err)`

Reply:
This would become a dead loop again.